### PR TITLE
Update to 7.10.0

### PR DESCRIPTION
**Pull request by RhCheGithubBot**
Tracking changes for fixing compatibility with upstream 7.10.0-SNAPSHOT. This PR was created automatically by Jenkins from job devtools-rh-che-rh-che-compatibility-test-dev.rdu2c.fabric8.io

### CODE REVIEW

**Comment by Katka92** [comment]
@ibuziuk The last PR check failed during a build on `Multi-Tenant Manager` part. Will you be able to take a look at that please?

**Comment by ibuziuk** [comment]
Provided fixes for Hosted Che specific fixes for https://github.com/eclipse/che/pull/16126

**Comment by ibuziuk** [comment]
@Katka92 now tests look broken 

> Could not find implementations for the following rules specified in the configuration:
>     label-undefined
>     no-constructor-vars
>     no-duplicate-key
>     no-trailing-comma
>     no-unreachable
> Try upgrading TSLint and/or ensuring that you have all the necessary custom rules installed.
> If TSLint was recently upgraded, you may have old rules configured which need to be cleaned up.
>         
> Fixed 1 error(s) in /tmp/rh-che/local_tests/inversify.config.ts

**Comment by Katka92** [comment]
Tests are failing becuase preferences handling via API is not copmatible with Che. The fix will be on Hosted Che side. I'll try to fix that during today/tomorrow.

**Comment by Katka92** [comment]
[test]

**Comment by Katka92** [comment]
@ibuziuk I've tried to fix the tests, but now the job is failing during the building of che on `Multi-Tenant Manager`. Could you please take a look?

**Comment by ibuziuk** [comment]
@Katka92 build is fixed, could you please take a look at failures?

**Comment by Katka92** [comment]
Failure of 2341: Application not available (Java Maven and Python Django)
Failure of 2342: Authorization token missing (Python Django)

**Comment by Katka92** [comment]
[test]

**Comment by Katka92** [comment]
The last run failed because of `Application not available`.

**Comment by Katka92** [comment]
[test]

**Comment by Katka92** [comment]
Test failed because of `Application not available` during workspace startup.

**Comment by Katka92** [comment]
[test]

**Comment by Katka92** [comment]
Last test failed to deploy Rh-Che.

**Comment by Katka92** [comment]
[test]

**Comment by Katka92** [comment]
The last failure is not clear for me. It can be an intermittent failure or test issue. I'll take a look at that.

**Comment by ibuziuk** [comment]
[test]

**Comment by ibuziuk** [comment]
[test]

**Comment by ibuziuk** [comment]
[test]

**Comment by ibuziuk** [comment]
@Katka92 looks like smth wrong with tests

**Comment by Katka92** [comment]
Yes, it seems like that. I've noticed that on Friday [1] and I hope I'll have time to look at it today. 

[1] https://github.com/redhat-developer/rh-che/pull/1805#issuecomment-598619349

**Comment by ibuziuk** [comment]
@ScrewTSW any update. What is the reason for test failure?
```2) Java Maven test
       Stop and remove workspace
         Delete workspace:
     UnexpectedAlertOpenError: unexpected alert open: {Alert text : }```

**Comment by ibuziuk** [comment]
[test]

**Comment by ScrewTSW** [comment]
[test]

**Comment by ibuziuk** [comment]
[test]

**Comment by ibuziuk** [comment]
[test]

**Comment by ScrewTSW** [comment]
[test]

**Comment by ibuziuk** [comment]
@ScrewTSW were you able to identify what is the issue with login to UD during the test execution:

> Login and wait dashboard
> Can not get bearer token. URL used: http://rhkeycloak-prcheck-1805.apps.che-dev.x6e0.p1.openshiftapps.com/auth/realms/che/protocol/openid-connect/token
> (node:196) UnhandledPromiseRejectionWarning: Error: Request failed with status code 503
>     at createError (/tmp/rh-che/local_tests/node_modules/axios/lib/core/createError.js:16:15)
>     at settle (/tmp/rh-che/local_tests/node_modules/axios/lib/core/settle.js:18:12)
>     at IncomingMessage.handleStreamEnd (/tmp/rh-che/local_tests/node_modules/axios/lib/adapters/http.js:192:11)
>     at emitNone (events.js:111:20)
>     at IncomingMessage.emit (events.js:208:7)
>     at endReadableNT (_stream_readable.js:1064:12)
>     at _combinedTickCallback (internal/process/next_tick.js:138:11)
>     at process._tickCallback (internal/process/next_tick.js:180:9)
> (node:196) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
> (node:196) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

File: dockerfiles/e2e-saas/Dockerfile [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # which accompanies this distribution, and is available at
 # http://www.eclipse.org/legal/epl-v10.html
 
-ARG TAG=latest
+ARG TAG=7.10.0
```
Comment (Katka92) [inline dockerfiles/e2e-saas/Dockerfile]: @ScrewTSW Why did you changed this argument setting? If you have no strong reason, please revert this change. This argument can be changed using `docker run` command.

Comment (ScrewTSW) [reply dockerfiles/e2e-saas/Dockerfile]: It's just for testing right now. in 7.10.0 PR the `latest` tag is scuffed, because the latest bits are 7.11.0 if I'm correct.

Comment (Katka92) [reply dockerfiles/e2e-saas/Dockerfile]: This is not true. PR check is using the version that is in a pom.xml [1]. So please revert those changes. Also, changes that should not be merged should have a clear message, not be hidden in some commit.
[1] https://github.com/redhat-developer/rh-che/blob/master/functional-tests/devscripts/run_tests.sh#L95

Comment (Katka92) [reply dockerfiles/e2e-saas/Dockerfile]: (and just a note - latest is currently 7.10.0 ;) )

Comment (ScrewTSW) [reply dockerfiles/e2e-saas/Dockerfile]: fixed, thank you for the clarification

**Comment by ScrewTSW** [comment]
[test]

**Comment by ScrewTSW** [comment]
[test]

**Comment by ScrewTSW** [comment]
[test]

**Comment by ScrewTSW** [comment]
[test]

**Comment by ScrewTSW** [comment]
Failed because of Application not available, retesting
[test]

**Comment by ScrewTSW** [comment]
[test]

**Comment by Katka92** [comment]
Test failed on `Application is not available`. Running test again.

**Comment by Katka92** [comment]
[test]

File: e2e-saas/utils/RhCheReporter.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+const driver: IDriver = rhcheContainer.get(TYPES.Driver);
+const driverHelper: DriverHelper = rhcheContainer.get(CLASSES.DriverHelper);
+const screenCatcher: ScreenCatcher = rhcheContainer.get(CLASSES.ScreenCatcher);
+let methodIndex: number = 0;
+let deleteScreencast: boolean = true;
+let testWorkspaceUtil: ITestWorkspaceUtil = rhcheContainer.get(TYPES.WorkspaceUtil);
+let preferencesHalder: PreferencesHandler = rhcheContainer.get(CLASSES.PreferencesHandler);
+
+class RhCheReporter extends mocha.reporters.Spec {
```
Comment (ibuziuk) [inline e2e-saas/utils/RhCheReporter.ts]: @Katka92 @ScrewTSW can we avoid using this reporter in 7.11.0 and fall back on upstream one ? e.g. remove this hot-fix for tests in the upstream for preference updates? cc: @rhopp

File: dockerfiles/e2e-saas/test_patches/Dashboard.patch [hunk 3]
```diff
@@ -1,13 +1,15 @@
---- ./pageobjects/dashboard/Dashboard.ts	2020-02-27 16:38:40.167786069 +0100
-+++ ./pageobjects/dashboard/Dashboard2.ts	2020-02-27 16:39:10.067174761 +0100
```
Comment (ibuziuk) [inline dockerfiles/e2e-saas/test_patches/Dashboard.patch]: @Katka92 @ScrewTSW are we going to remove this patch for 7.11.0 ?

Comment (ScrewTSW) [reply dockerfiles/e2e-saas/test_patches/Dashboard.patch]: You are commenting on an old file right now, but anyways, if the objects are removed from upstream in 7.11.0 we should be able to get rid of the patch, yes.

Comment (ibuziuk) [reply dockerfiles/e2e-saas/test_patches/Dashboard.patch]: @ScrewTSW could you please make sure that we will be not dependent on the UD menu - https://github.com/eclipse/che/issues/16165

File: e2e-saas/utils/RhCheReporter.ts (hunk 2 above)
Comment (ScrewTSW) [reply e2e-saas/utils/RhCheReporter.ts]: I'm not sure I can answer this. @Katka92 has better knowledge of the reporter classes.

File: dockerfiles/e2e-saas/test_patches/Dashboard.patch (hunk 3 above)
Comment (ScrewTSW) [reply dockerfiles/e2e-saas/test_patches/Dashboard.patch]: I can look into that next week

File: dockerfiles/e2e-saas/test_patches/CheApiRequestHandler.patch [hunk 4]
```diff
@@ -0,0 +1,56 @@
+diff --git a/tests/e2e/utils/requestHandlers/CheApiRequestHandler.ts b/tests/e2e/utils/requestHandlers/CheApiRequestHandler.ts
```
Comment (ibuziuk) [inline dockerfiles/e2e-saas/test_patches/CheApiRequestHandler.patch]: We should get rid of this path in the upcoming upstream releases
@Katka92 @ScrewTSW could you please create an issue for that and raise with @rhopp during prioritization / planning

File: dockerfiles/e2e-saas/docker-entrypoint.sh [hunk 5]
```diff
@@ -139,10 +143,12 @@ export DISPLAY=:1.0
 
 echo "Applying patches..."
 # currently in /tmp/rh-che folder
-patch -u ../e2e/pageobjects/dashboard/Dashboard.ts -i ../test_patches/dashboard.patch 
+patch -u ../e2e/pageobjects/dashboard/Dashboard.ts -i ../test_patches/Dashboard.patch
+patch -u ../e2e/utils/requestHandlers/CheApiRequestHandler.ts -i ../test_patches/CheApiRequestHandler.patch
+patch -u ../e2e/package.json -i ../test_patches/upstream_package.patch
```
Comment (ibuziuk) [inline dockerfiles/e2e-saas/docker-entrypoint.sh]: we should get rid of all the downstream patches. Could you please create an issue for that?

File: e2e-saas/utils/RhCheReporter.ts (hunk 2 above)
Comment (ibuziuk) [reply e2e-saas/utils/RhCheReporter.ts]: the goal to get rid of downstream UD patches, custom reporters etc. We should fully reuse upstream tests

Comment (Katka92) [reply e2e-saas/utils/RhCheReporter.ts]: @ibuziuk The main problem which RhCheReporter is solving is that if we use upstream reporter, the inversion of control uses all upstream classes, even for obtaining user token. We need to inject the RhCheSpecific classes prior to anything else happens, and the first class that is executed during the test run is the reporter. So, this is not just a hot-fix for preferences handler but also fix for all API calls using a token on the rh-che side (like setting up a preferences).

File: dockerfiles/e2e-saas/docker-entrypoint.sh (hunk 5 above)
Comment (Katka92) [reply dockerfiles/e2e-saas/docker-entrypoint.sh]: I've created an issue https://github.com/eclipse/che/issues/16456 but I'm not sure if we will be able to handle that, as the release should start after 2 working days. I'll do my best.

File: dockerfiles/e2e-saas/test_patches/CheApiRequestHandler.patch (hunk 4 above)
Comment (Katka92) [reply dockerfiles/e2e-saas/test_patches/CheApiRequestHandler.patch]: It will be part of https://github.com/eclipse/che/issues/16456.

File: dockerfiles/e2e-saas/docker-entrypoint.sh (hunk 5 above)
Comment (ibuziuk) [reply dockerfiles/e2e-saas/docker-entrypoint.sh]: @Katka92 no rush, just please prioritize it for 7.12

File: dockerfiles/e2e-saas/docker-entrypoint.sh [hunk 6]
```diff
@@ -29,6 +28,9 @@ fi
 if [ -z $TEST_SUITE ]; then
   TEST_SUITE="test-java-maven"
```
Comment (Katka92) [inline dockerfiles/e2e-saas/docker-entrypoint.sh]: This test suite was removed. It should be replaced with `test-all` suite.
